src/lightgbm/src/main/scala/LightGBMBooster.scala

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark
+
+import com.microsoft.ml.lightgbm.{CSRUtils, SWIGTYPE_p_void, lightgbmlib, lightgbmlibConstants}
+import com.microsoft.ml.spark.LightGBMUtils.{intToPtr, newDoubleArray, newIntArray}
+import org.apache.spark.ml.linalg.{DenseVector, SparseVector, Vector}
+
+/** Represents a LightGBM Booster learner
+  * @param model The string serialized representation of the learner
+  */
+class LightGBMBooster(val model: String) extends Serializable {
+  /**
+    * Transient variable containing local machine's pointer to native booster
+    */
+  @transient
+  var boosterPtr: SWIGTYPE_p_void = null
+
+  def numClasses(): Int = {
+    if (boosterPtr == null) {
+      LightGBMUtils.initializeNativeLibrary()
+      boosterPtr = getModel()
+    }
+    val numClasses = lightgbmlib.new_intp()
+    LightGBMUtils.validate(lightgbmlib.LGBM_BoosterGetNumClasses(boosterPtr, numClasses), "Booster GetNumClasses")
+    lightgbmlib.intp_value(numClasses)
+  }
+
+  def scoreRaw(features: Vector): Double = {
+    // Reload booster on each node
```

Comment:
I think we can eventually optimize scoring more by doing it in batch instead of on a row by row basis (in order to reduce the translation costs from Java to C++), although I'm honestly not sure how big the performance gains would be, since scoring seems to be quite fast already